cpg_workflows/inputs.py

```diff
@@ -66,18 +67,18 @@ def create_multicohort() -> MultiCohort:
         # TODO (mwelland): ensures the get-analyses-in-dataset query is only made once per Dataset
         _populate_cohort(cohort, sgs_by_dataset_for_cohort, read_pedigree=read_pedigree)
 
-    # TODO: (vivbak) Remove this broken block of code to unblock. Temporary measure. Will fix asap.
-
-    # # now populate the datasets uniquely in the multicohort, each containing all sequencing groups in this dataset
-    # # the MultiCohort has a dictionary of {dataset_name: Dataset} objects
-    # # each of those Dataset objects links to all SequencingGroups in that Dataset, across all Cohorts
-    # for cohort in multicohort.get_cohorts():
-    #     for dataset in cohort.get_datasets():
-    #         # is this dataset already in the multicohort, if so use it, otherwise add this one
-    #         mc_dataset = multicohort.add_dataset(dataset)
-    #         for sg in dataset.get_sequencing_groups():
-    #             # add each SG object to the MultiCohort level Dataset
-    #             mc_dataset.add_sequencing_group_object(sg)
+    # now populate the datasets uniquely in the multicohort, each containing all sequencing groups in this dataset
+    # the MultiCohort has a dictionary of {dataset_name: Dataset} objects
+    # each of those Dataset objects links to all SequencingGroups in that Dataset, across all Cohorts
+    for cohort in multicohort.get_cohorts():
+        for dataset in cohort.get_datasets():
+            # is this dataset already in the multicohort, if so use it, otherwise add this one
+            # Since datasets are created on the Cohort, and are added to the multicohort, we need to
+            # perform a deepcopy to avoid manipulating the cohort dataset at this point
+            mc_dataset = multicohort.add_dataset(copy.deepcopy(dataset))
```

Comment:
After our session, I am thinking this deep copy should be here, it would speed up the function as it only created copy when needed.

https://github.com/populationgenomics/production-pipelines/blob/20da8c3b8939b434eeeed404c80bdb7d8d79fdf7/cpg_workflows/targets.py#L191